python/lsst/pipe/tasks/dcrAssembleCoadd.py

```diff
@@ -762,6 +779,30 @@ def calculateGain(self, modelIter, baseGain=1.):
             A value of 1.0 gives equal weight to both solutions.
         """
         if self.config.useProgressiveGain:
-            iterGain = np.log(modelIter) if modelIter > 0 else baseGain
-            return max(baseGain, iterGain)
-        return baseGain
+            iterGain = np.log(modelIter)*self.config.baseGain if modelIter > 0 else self.config.baseGain
+            return max(self.config.baseGain, iterGain)
+        return self.config.baseGain
+
+    def calculateModelWeights(self, maskedImage):
+        """Build an array that smoothly tapers to 0 away from detected sources.
+
+        Parameters
+        ----------
+        maskedImage : `numpy.ndarray`
+            The input masked image to calculate weights for.
+
+        Returns
+        -------
+        weights : `numpy.ndarray` or `float`
+            A 2D array of weight values that tapers smoothly to zero away from detected sources.
+            Or a placeholder value of 1.0 if ``self.config.modelWidth`` is set to a negative number.
+        """
+        if self.config.modelWidth < 0:
+            return 1.
```

Comment:
Again, I don't like this sneaky behavior. If you want a fallback option to have all the weights be 1, that should be an explicit option, not something that happens if the width is set negative.